create-publication-from-scopus/src/main/java/no/sikt/nva/scopus/ScopusHandler.java

```diff
@@ -86,6 +93,54 @@ private Reference generateReference(DocTp docTp) {
         return reference;
     }
 
+    private List<Contributor> generateContributors(DocTp docTp) {
+        List<Contributor> contributors = new ArrayList<>();
+        extractAuthorGroup(docTp).forEach(authorGroupTp -> contributors.addAll(generateContributorsFromAuthorGroup(authorGroupTp)));
+        return contributors;
+    }
+
+    private List<Contributor> generateContributorsFromAuthorGroup(AuthorGroupTp authorGroupTp){
+        List<Contributor> contributors = new ArrayList<>();
+        authorGroupTp
+            .getAuthorOrCollaboration()
+            .forEach(authorOrCollaboration ->
+                         contributors.add(generateContributorFromAuthorOrCollaboration(authorOrCollaboration)));
+        return contributors;
+    }
+
+    private List<AuthorGroupTp> extractAuthorGroup(DocTp docTp) {
+        return docTp.getItem().getItem().getBibrecord().getHead().getAuthorGroup();
+    }
+
+    private Contributor generateContributorFromAuthorOrCollaboration(Object authorOrCollaboration) {
+        return authorOrCollaboration instanceof AuthorTp
+                   ? generateContributorFromAuthorTp((AuthorTp) authorOrCollaboration)
+                   : generateContributorFromCollaborationTp(
+                       (CollaborationTp) authorOrCollaboration);
+    }
+
+    private Contributor generateContributorFromAuthorTp(AuthorTp author) {
+        var sequence = Integer.parseInt(author.getSeq());
+        var identity = new Identity();
+        identity.setName(determineContributorName(author));
+        return new Contributor(identity, null, null, sequence, false);
+    }
+
+    private Contributor generateContributorFromCollaborationTp(CollaborationTp collaboration) {
+        var sequence = Integer.parseInt(collaboration.getSeq());
+        var identity = new Identity();
+        identity.setName(determineContributorName(collaboration));
+        return new Contributor(identity, null, null, sequence, false);
+    }
+
+    private String determineContributorName(AuthorTp author) {
```

Comment:
I følge dokumentasjon så er preferred name optional, men etter å ha kjørt over 25000+ test data, men python script så har jeg ikke funnet en eneste author uten preferred name

Reply:
Good enough for me.